java/memory/memory-core/src/main/java/org/apache/arrow/memory/BoundsChecking.java

```diff
@@ -54,7 +55,7 @@
       unsafeFlagValue = envProperty;
     }
 
-    BOUNDS_CHECKING_ENABLED = !"true".equals(unsafeFlagValue);
+    BOUNDS_CHECKING_ENABLED = "false".equals(unsafeFlagValue);
```

Comment:
I think it'd clearer this way, i.e., the original without the leading `!`. By default, we want the bounds checking disabled.

```suggestion
    BOUNDS_CHECKING_ENABLED = "true".equals(unsafeFlagValue);
```

There is a test in `java/memory/memory-core/src/test/java/org/apache/arrow/memory/TestBoundaryChecking.java` (`testDefaultValue`) that checks this. It should  still pass after this change

Reply:
The `TestBoundaryChecking` still passes after this change https://github.com/sirensolutions/arrow/pull/16/commits/831c485d90a0d5f76cd903ce7f7800a9c63dca1f